addons/stock/models/stock_orderpoint.py

```diff
@@ -342,81 +342,65 @@ def _get_orderpoint_action(self):
         orderpoints = orderpoints - orderpoints_removed
         to_refill = defaultdict(float)
         all_product_ids = []
-        all_warehouse_ids = []
-        # Take 3 months since it's the max for the forecast report
-        to_date = add(fields.date.today(), months=3)
-        qty_by_product_warehouse = self.env['report.stock.quantity']._read_group(
-            [('date', '=', to_date), ('state', '=', 'forecast')],
-            ['product_id', 'product_qty', 'warehouse_id'],
-            ['product_id', 'warehouse_id'], lazy=False)
-        for group in qty_by_product_warehouse:
-            warehouse_id = group.get('warehouse_id') and group['warehouse_id'][0]
-            if group['product_qty'] >= 0.0 or not warehouse_id:
+        all_replenish_location_ids = self.env['stock.location'].search([('replenish_location', '=', True)])
+        ploc_per_day = {}
+        qty_by_product = self.env['report.stock.quantity']._read_group(
+            [('state', '=', 'forecast')],
+            ['product_id', 'product_qty'],
+            ['product_id'], lazy=False)
```

Comment:
I think you should not use the report.stock.quantity. Instead do a read group.
- to_refill should be a key (product, location) now
- First step do a loop for each location checking the `virtual_available` for all the products
- Second step, for each missing key (product, location) -> compute the lead time and the real `virtual_available` to now + lead times
-,Remove the quantity in progress
- Create the orderpoint